workflows/bamtofastq.nf

```diff
@@ -65,22 +94,121 @@ workflow BAMTOFASTQ {
 
     ch_versions = Channel.empty()
 
-    //
-    // SUBWORKFLOW: Read in samplesheet, validate and stage input files
-    //
-    INPUT_CHECK (
-        ch_input
+    // SUBWORKFLOW: Prepare indices bai/crai/fai if not provided
+    PREPARE_INDICES(
```

Comment:
This subworkflow is only run if index files are not provided. It checks if the index files are there and if not, it will only run on those samples for which index files were not added to the samplesheet.